admin.go

```diff
@@ -0,0 +1,162 @@
+package sarama
+
+type ClusterAdmin interface {
+	CreateTopic(topic string, detail *TopicDetail) error
+	DeleteTopic(topic string) error
+	CreatePartitions(topic string, count int32, assignment [][]int32, validateOnly bool) error
+	DeleteRecords(topic string, partitionOffsets map[int32]int64) error
+	DescribeConfig(resource ConfigResource) ([]ConfigEntry, error)
+	AlterConfig(resourceType ConfigResourceType, name string, entries map[string]*string, validateOnly bool) error
+	CreateAcl(resource Resource, acl Acl) error
+	ListAcls(filter AclFilter) ([]ResourceAcls, error)
+	DeleteAcl(filter AclFilter, validateOnly bool) ([]MatchingAcl, error)
+}
+
+type clusterAdmin struct {
+	client Client
+	conf   *Config
+}
+
+// NewClusterAdmin creates a new ClusterAdmin using the given broker addresses and configuration.
+func NewClusterAdmin(addrs []string, conf *Config) (ClusterAdmin, error) {
+	client, err := NewClient(addrs, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check that we are not dealing with a closed Client before processing any other arguments
+	if client.Closed() {
```

Comment:
unnecessary since you literally just created it

Reply:
Will delete that